client/blocks/domain-tip/index.jsx

```diff
@@ -55,6 +59,20 @@ class DomainTip extends Component {
 		);
 	}
 
+	getDomainUpsellNudgeText() {
+		const siteHas100YearPlan = this.props.site?.plan?.product_slug === PLAN_100_YEARS;
```

Comment:
We should probably add this to 

https://github.com/Automattic/wp-calypso/blob/8a8d542c4012529dfe3ec2d0d7c9db6d59bd3073/packages/calypso-products/src/main.ts#L311-L322

You can actually feed the plan object to this call :)